docs/addons/nginx.md

```diff
@@ -0,0 +1,135 @@
+# NGINX Add-on
+
+This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ingress-nginx/deploy/) on Amazon EKS. NGINX ingress controller is using [NGINX](https://www.nginx.org/) as a reverse proxy and load balancer. 
+
+Other than handling Kubernetes ingress objects, this ingress controller can facilitate multi-tenancy and segregation of workload ingresses based on host name (host-based routing) and/or URL Path (path based routing). 
+
+## Usage
+
+NGINX add-on is a standard SSP add-on and it is usage is similar to the rest of the add-ons.
+
+```typescript
+import { NginxAddOn, ClusterAddOn, EksBlueprint }  from '@shapirov/cdk-eks-blueprint';
+
+const subdomain  = ...;
+const nginxAddOn = new NginxAddOn(({ 
+    internetFacing: true, 
+    backendProtocol: "tcp", 
+    externaDnsHostname: subdomain, 
+    crossZoneEnabled: false 
+}));
+
+const addOns: Array<ClusterAddOn> = [ nginxAddOn ];
+
+const app = new cdk.App();
+new EksBlueprint(app, 'my-stack-name', addOns, [], {
+  env: {    
+      account: <AWS_ACCOUNT_ID>,
+      region: <AWS_REGION>,
+  },
+});
+```
+
+To validate that installation is successful run the following command:
+
+```bash
+$ kubectl get po -n kube-system
+NAME                                                              READY   STATUS    RESTARTS   AGE
+ssp-addon-nginx-ingress-78b8567p4q6   1/1     Running   0          4d10h
+```
+
+Note that the ingress is deployed in the `kube-system` namespace.
+
+Once deployed, it allows applications to create ingress objects and use host based routing with external DNS support, if External DNS Add-on is installed.
+
+## Configuration
+
+ - `backendProtocol`: indication for AWS Load Balancer controller with respect to the protocol supported on the load balancer. TCP by default.
```

Comment:
should we make these tables? You have different formatting for configuration in the ExternalDNSAddon